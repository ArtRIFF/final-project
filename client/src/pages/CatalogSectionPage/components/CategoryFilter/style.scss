@import '../../../../global_scss/vars.scss';
@import '../../../../global_scss/mixins.scss';

.category-filter {
  @include flex-kit(space-between);
  width: inherit;
  height: inherit;

  font-family: $main-font__regular;
  @include font-value(300, 20px, 25px);
  
  &__result {
    @media (max-width: 460px) {
      display: none;
  }
  }

  &--btn {
    display: none;
    @media (max-width:$desktop-size) {
      display: inline-block;
  }
  }

  select {
    background-color: transparent;
    border: none;
    font-family: inherit;
    font-size: inherit;
    line-height: inherit;
    outline: none;
    cursor: pointer;
    appearance: none;
    background-image: url('../../icons/arrow-down.svg') ;
    background-position: right;
    background-repeat: no-repeat;
    padding-right: 25px;
  }

}