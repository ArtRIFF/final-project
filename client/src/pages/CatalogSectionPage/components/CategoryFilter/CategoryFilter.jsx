import './style.scss';
import ButtonAll from '../../../../components/Button/ButtonAll/ButtonAll';
import PropTypes from 'prop-types';

const CategoryFilter = ({ onClickFunc, setResult, filterRequest }) => {

  const sortByNewest = () => {
    filterRequest("sort=NEW");
  }

  const sortByBestseller = () => {
    filterRequest("sort=BESTSELLER");
  }

  const sortByPrice = () => {
    filterRequest("sort=currentPrice");
  }

  const onSelectorChange = (e) => {
    switch (e.target.value) {
      case "price":
        sortByPrice();
        break;
      case "newest":
        sortByNewest();
        break;
      case "bestseller":
        sortByBestseller();
        break;
    }
  };
  return (
    <div className="category-filter">
      <div onClick={onClickFunc} className='category-filter--btn'>
        <ButtonAll text='Filter' className='section__btn-subscribe' />
      </div>
      <div className='category-filter__result'>Results: <span>{setResult}</span></div>
      <select data-testid='select-sortBy' onChange={onSelectorChange} name="select-sortBy" defaultValue={'DEFAULT'} >
        <option value="DEFAULT" disabled>Sort by</option>
        <option value="price">Price</option>
        <option value="newest">Newest</option>
        <option value="bestseller">Bestseller</option>
      </select>
    </div>
  );
};

CategoryFilter.propTypes = {
  onClickFunc: PropTypes.func,
  setResult: PropTypes.number,
  filterRequest: PropTypes.func
};

CategoryFilter.defaultProps = {
  onClickFunc: null
};

export default CategoryFilter;