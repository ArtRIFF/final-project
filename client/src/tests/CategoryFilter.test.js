import CategoryFilter from '../pages/CatalogSectionPage/components/CategoryFilter/CategoryFilter'
import '@testing-library/jest-dom';
import { render, cleanup, screen, fireEvent} from "@testing-library/react";
import renderer from 'react-test-renderer';

afterEach(cleanup);

const onClickFunc = jest.fn();
const filterRequest = jest.fn();

const renderComponent = () => {
  return (render(<CategoryFilter
    onClickFunc={onClickFunc}
    setResult={10}
    filterRequest={filterRequest}
    allCollectionArray={[
      {
        statusProduct: 'NEW',
        currentPrice: 12
      },
      {
        statusProduct: 'BESTSELLER',
        currentPrice: 6
      },
      {
        statusProduct: 'BESTSELLER',
        currentPrice: 3
      }

    ]}
    hasAnyFilters={true}
  />))
}

it("renders with snapshot", () => {
  const categoryFilter = renderer.create(<CategoryFilter
    onClickFunc={onClickFunc}
    setResult={10}
    filterRequest={filterRequest}
    allCollectionArray={[]}
  />).toJSON()
  expect(categoryFilter).toMatchSnapshot();
});

it("renders first result from arrayCollection", () => {
  renderComponent();
  expect(screen.getByText("3")).toBeInTheDocument();
});

it("test onclick event", () => {
  renderComponent();
  fireEvent.click(screen.getByText(/Filter/i))
  expect(onClickFunc).toHaveBeenCalled();
});

it("test onchange event", () => {
  renderComponent();
  const selectElem = screen.getByTestId("select-sortBy");
  fireEvent.change(selectElem, { target: { value: 'price' } });
  expect(filterRequest).toHaveBeenCalledTimes(1);
});