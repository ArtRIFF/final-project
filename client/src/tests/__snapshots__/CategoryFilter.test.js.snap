// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with snapshot 1`] = `
<div
  className="category-filter"
>
  <div
    className="category-filter--btn"
    onClick={[MockFunction]}
  >
    <div>
      <button
        className="section__btn-subscribe"
      >
        Filter
      </button>
    </div>
  </div>
  <div
    className="category-filter__result"
  >
    Results: 
    <span>
      0
    </span>
  </div>
  <select
    data-testid="select-sortBy"
    defaultValue="DEFAULT"
    name="select-sortBy"
    onChange={[Function]}
  >
    <option
      disabled={true}
      value="DEFAULT"
    >
      Sort by
    </option>
    <option
      value="price"
    >
      Price
    </option>
    <option
      value="newest"
    >
      Newest
    </option>
    <option
      value="bestseller"
    >
      Bestseller
    </option>
  </select>
</div>
`;
